boa/src/syntax/parser/expression/primary/object_initializer/mod.rs

```diff
@@ -82,7 +82,7 @@ impl TokenParser for ObjectLiteral {
             }
         }
 
-        Ok(Node::object(elements))
+        Ok(Object::new(elements).into())
```

Comment:
I would suggest that this returns an `Object`, instead of a Node. We should only convert to a `Node` once it's required.

Reply:
Won't it need to be returned as a Node because of the type of ParseResult? (<Node, ParseError>)

Reply:
> Won't it need to be returned as a Node because of the type of ParseResult? (<Node, ParseError>)

We can change that by setting the `Output` to an `Object`, and then changing the function signature :)

Reply:
Done